data/kernels/basic.cl

```diff
@@ -614,11 +614,54 @@ colorin_clipping (read_only image2d_t in, write_only image2d_t out, const int wi
   write_imagef (out, (int2)(x, y), pixel);
 }
 
+float
+rgb_norm_vect (float4 rgb, int rgb_norm, float norm_exp)
+{
+  switch(rgb_norm)
```

Comment:
not sure about `case 2` and `case 3` since they are particular cases of `case 4` (with values of `norm_exp` nice enough to avoid slow calls to `pow`). I would just merge them into `case 4` and check the `norm_exp` to branch, or even completely ignore them in OpenCL since branches can make it super slow and it doesn't mind the load (so you just use the general case 4).

Reply:
removed case 2 and 3. In tonecurve (openmp) detection of exp =1 or 2 and use of simplified formulas.